tests/json/eq.rs

```diff
@@ -6,6 +6,7 @@ pub trait NumEq {
     fn int_eq(&self, i: i64) -> bool;
 }
 
+#[cfg_attr(feature = "clippy", allow(cast_precision_loss))]
```

Comment:
Slightly tangential, but this has been on my radar for a while: we're currently using the `nalgebra` crate to handle floating-point equality, but that's a fairly heavyweight dependency for such a small task. There are several crates for comparing floating points that aren't part of entire linear algebra toolkits; from a quick search, I've found [`approx`](https://crates.io/crates/approx) and [`float-cmp`](https://crates.io/crates/float-cmp). We should look into replacing `nalgebra` with something more fitting at some point.